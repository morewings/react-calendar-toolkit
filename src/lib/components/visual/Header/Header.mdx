---
name: Header
menu: Customize UI
---
import { Playground, Props } from 'docz';
import {Datepicker} from 'lib';
import Header from './Header';

`Header` component is displayed on top of `Datepicker`. It receives today and selected Date objects. Replace it with custom component.

<Playground>
  {() => {
    const style = {
      color: 'black',
      textAlign: 'center',
      padding: '16px 0',
      border: '2px dashed pink',
      margin: '2px'
    };
    const CustomComponent = ({selectedDate, title, todayDate}) => {
      return (
        <div style={style}>
          {title}
        </div>
      )
    };
    return (
      <Datepicker
        onDateSet={date => {console.log('date set', date);}}
        minPrecision="month"
        renderHeaderAs={CustomComponent}
        title="Demo datepicker" />
    )
  }}
</Playground>

## Props
<Props of={Header} />

