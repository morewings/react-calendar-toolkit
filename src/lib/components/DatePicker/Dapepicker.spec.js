import React from 'react';
import {render} from '@testing-library/react';
import {DatepickerMockProvider as Provider} from 'lib/utils/testProvider';
import Calendar from 'lib/components/logic/Calendar';
import WeekDays from 'lib/components/logic/Weekdays';
import Selector from 'lib/components/logic/Selector';
import Header from 'lib/components/logic/Header';
import DatePicker from './DatePicker';

jest.mock('lib/components/logic/Calendar', () => ({
  __esModule: true,
  default: jest.fn(() => <div>Calendar</div>),
}));

jest.mock('lib/components/logic/Weekdays', () => ({
  __esModule: true,
  default: jest.fn(() => <div>Weekdays</div>),
}));

jest.mock('lib/components/logic/Selector', () => ({
  __esModule: true,
  default: jest.fn(() => <div>Selector</div>),
}));

jest.mock('lib/components/logic/Header', () => ({
  __esModule: true,
  default: jest.fn(() => <div>Header</div>),
}));

const MockComponent = jest.fn(() => <div>MockComponent</div>);

const MockWrapper = jest.fn(({children}) => (
  <div data-testid="wrapper">{children}</div>
));

const onDateSet = jest.fn();
const date = {
  base: new Date(2020, 0, 12), // 12.01.2020
  plusOneDay: new Date(2020, 0, 13),
  minusOneDay: new Date(2020, 0, 11),
  minusOneMonth: new Date(2019, 11, 12),
  plusOneMonth: new Date(2020, 1, 12),
  plusOneYear: new Date(2021, 1, 12),
  minusOneYear: new Date(2019, 1, 12),
};

const renderWithProps = (props = {}) =>
  render(
    <DatePicker
      {...props}
      today={new Date(2020, 0, 10)}
      onDateSet={onDateSet}
    />,
    {
      wrapper: ({children}) => <Provider>{children}</Provider>,
    }
  );

const scrollIntoView = jest.fn();

describe('DatePicker', () => {
  beforeEach(() => {
    onDateSet.mockClear();
    Header.mockClear();
    Calendar.mockClear();
    Selector.mockClear();
    WeekDays.mockClear();
  });

  beforeAll(() => {
    // eslint-disable-next-line fp/no-mutation
    window.HTMLElement.prototype.scrollIntoView = scrollIntoView;
  });

  afterAll(() => {
    // eslint-disable-next-line fp/no-mutation
    window.HTMLElement.prototype.scrollIntoView = undefined;
  });

  it('renders', () => {
    const {asFragment} = renderWithProps();
    expect(asFragment()).toMatchSnapshot();
    expect(Header.mock.calls[0][0]).toMatchSnapshot();
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
    expect(Selector.mock.calls[0][0]).toMatchSnapshot();
  });

  it('renders without header', () => {
    const {asFragment} = renderWithProps({showHeader: false});
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders title', () => {
    renderWithProps({title: 'Title'});
    expect(Header.mock.calls[0][0]).toMatchSnapshot();
  });

  it('disables weekend highlight', () => {
    renderWithProps({highlightWeekends: false});
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
  });

  it('passes `disableDate`', () => {
    const disableDate = jest.fn();
    renderWithProps({disableDate});
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
  });

  it('passes `highlightDate`', () => {
    const highlightDate = jest.fn();
    renderWithProps({highlightDate});
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
  });

  describe.each`
    initialDate  | startDate            | endDate             | today              | minPrecision
    ${date.base} | ${date.minusOneYear} | ${date.plusOneYear} | ${date.plusOneDay} | ${'day'}
    ${date.base} | ${date.minusOneYear} | ${date.plusOneYear} | ${date.plusOneDay} | ${'month'}
    ${date.base} | ${date.minusOneYear} | ${date.plusOneYear} | ${date.plusOneDay} | ${'year'}
  `(
    'renders date related props:',
    ({initialDate, startDate, endDate, today, minPrecision}) => {
      it(`initialDate: ${initialDate}, startDate: ${startDate}, endDate: ${endDate}, today: ${today}, minPrecision: ${minPrecision} `, () => {
        const {asFragment} = renderWithProps({
          initialDate,
          startDate,
          endDate,
          today,
          minPrecision,
        });
        expect(asFragment()).toMatchSnapshot();
        expect(Header.mock.calls[0][0]).toMatchSnapshot();
        expect(Selector.mock.calls[0][0]).toMatchSnapshot();
        expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
        minPrecision === 'day' &&
          expect(WeekDays.mock.calls[0][0]).toMatchSnapshot();
      });
    }
  );

  it('renders Header custom UI', () => {
    const {asFragment} = renderWithProps({renderHeaderAs: MockComponent});
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Selector custom UI', () => {
    const {asFragment} = renderWithProps({renderSelectorAs: MockComponent});
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders custom Wrapper', () => {
    const {getByTestId} = renderWithProps({wrapWith: MockWrapper});
    expect(getByTestId('wrapper')).toMatchSnapshot();
  });

  it('passes Day(s) Calendar custom UI', () => {
    renderWithProps({
      renderDayAs: 'MockComponent',
      wrapDaysWith: 'MockWrapper',
    });
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
  });

  it('passes Month(s) Calendar custom UI', () => {
    renderWithProps({
      renderMonthAs: 'MockComponent',
      wrapMonthWith: 'MockWrapper',
    });
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
  });

  it('passes Year(s) Calendar custom UI', () => {
    renderWithProps({
      renderYearAs: 'MockComponent',
      wrapYearWith: 'MockWrapper',
    });
    expect(Calendar.mock.calls[0][0]).toMatchSnapshot();
  });

  it('passes Weekday(s) Row custom UI', () => {
    renderWithProps({
      renderWeekDayAs: 'MockComponent',
      wrapWeekDaysWith: 'MockWrapper',
      minPrecision: 'day',
    });
    expect(WeekDays.mock.calls[0][0]).toMatchSnapshot();
  });
});
