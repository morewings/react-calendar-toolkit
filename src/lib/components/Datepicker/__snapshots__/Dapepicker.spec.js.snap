// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker disables weekend highlight 1`] = `
Object {
  "disableDate": [Function],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [Function],
  "highlightWeekends": false,
  "onDateSet": [Function],
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker passes \`disableDate\` 1`] = `
Object {
  "disableDate": [MockFunction],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [Function],
  "highlightWeekends": true,
  "onDateSet": [Function],
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker passes \`highlightDate\` 1`] = `
Object {
  "disableDate": [Function],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [MockFunction],
  "highlightWeekends": true,
  "onDateSet": [Function],
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker passes Day(s) Calendar custom UI 1`] = `
Object {
  "disableDate": [Function],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [Function],
  "highlightWeekends": true,
  "onDateSet": [Function],
  "precision": "day",
  "renderAs": "MockComponent",
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "foo",
}
`;

exports[`Datepicker passes Month(s) Calendar custom UI 1`] = `
Object {
  "disableDate": [Function],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [Function],
  "onDateSet": [Function],
  "precision": "month",
  "renderAs": "MockComponent",
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "foo",
}
`;

exports[`Datepicker passes Weekdays Row custom UI 1`] = `
Object {
  "names": Array [
    Object {
      "abbreviated": "Sun",
      "narrow": "S",
      "numeric": 0,
      "short": "Su",
      "wide": "Sunday",
    },
    Object {
      "abbreviated": "Mon",
      "narrow": "M",
      "numeric": 1,
      "short": "Mo",
      "wide": "Monday",
    },
    Object {
      "abbreviated": "Tue",
      "narrow": "T",
      "numeric": 2,
      "short": "Tu",
      "wide": "Tuesday",
    },
    Object {
      "abbreviated": "Wed",
      "narrow": "W",
      "numeric": 3,
      "short": "We",
      "wide": "Wednesday",
    },
    Object {
      "abbreviated": "Thu",
      "narrow": "T",
      "numeric": 4,
      "short": "Th",
      "wide": "Thursday",
    },
    Object {
      "abbreviated": "Fri",
      "narrow": "F",
      "numeric": 5,
      "short": "Fr",
      "wide": "Friday",
    },
    Object {
      "abbreviated": "Sat",
      "narrow": "S",
      "numeric": 6,
      "short": "Sa",
      "wide": "Saturday",
    },
  ],
}
`;

exports[`Datepicker passes Year(s) Calendar custom UI 1`] = `
Object {
  "disableDate": [Function],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [Function],
  "onDateSet": [Function],
  "precision": "year",
  "renderAs": "MockComponent",
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "foo",
}
`;

exports[`Datepicker renders 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Weekdays
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders 2`] = `
Object {
  "renderAs": [Function],
  "title": "",
}
`;

exports[`Datepicker renders 3`] = `
Object {
  "disableDate": [Function],
  "endDate": 2020-02-25T00:00:00.000Z,
  "highlightDate": [Function],
  "highlightWeekends": true,
  "onDateSet": [Function],
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker renders 4`] = `
Object {
  "endDate": 2020-02-25T00:00:00.000Z,
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578268800000,
  "startDate": 2020-01-01T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578268800000,
}
`;

exports[`Datepicker renders Header custom UI 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Weekdays
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders Selector custom UI 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Weekdays
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders custom Wrapper 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="foo"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Weekdays
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: day  1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Weekdays
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: day  2`] = `
Object {
  "renderAs": [Function],
  "title": "",
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: day  3`] = `
Object {
  "endDate": 2021-02-12T00:00:00.000Z,
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578787200000,
  "startDate": 2019-02-12T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578787200000,
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: day  4`] = `
Object {
  "disableDate": [Function],
  "endDate": 2021-02-12T00:00:00.000Z,
  "highlightDate": [Function],
  "highlightWeekends": true,
  "onDateSet": [Function],
  "precision": "day",
  "renderAs": [Function],
  "selectedTimestamp": 1578787200000,
  "startDate": 2019-02-12T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578787200000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: day  5`] = `
Object {
  "names": Array [
    Object {
      "abbreviated": "Sun",
      "narrow": "S",
      "numeric": 0,
      "short": "Su",
      "wide": "Sunday",
    },
    Object {
      "abbreviated": "Mon",
      "narrow": "M",
      "numeric": 1,
      "short": "Mo",
      "wide": "Monday",
    },
    Object {
      "abbreviated": "Tue",
      "narrow": "T",
      "numeric": 2,
      "short": "Tu",
      "wide": "Tuesday",
    },
    Object {
      "abbreviated": "Wed",
      "narrow": "W",
      "numeric": 3,
      "short": "We",
      "wide": "Wednesday",
    },
    Object {
      "abbreviated": "Thu",
      "narrow": "T",
      "numeric": 4,
      "short": "Th",
      "wide": "Thursday",
    },
    Object {
      "abbreviated": "Fri",
      "narrow": "F",
      "numeric": 5,
      "short": "Fr",
      "wide": "Friday",
    },
    Object {
      "abbreviated": "Sat",
      "narrow": "S",
      "numeric": 6,
      "short": "Sa",
      "wide": "Saturday",
    },
  ],
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: month  1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: month  2`] = `
Object {
  "renderAs": [Function],
  "title": "",
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: month  3`] = `
Object {
  "endDate": 2021-02-12T00:00:00.000Z,
  "precision": "month",
  "renderAs": [Function],
  "selectedTimestamp": 1578787200000,
  "startDate": 2019-02-12T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578787200000,
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: month  4`] = `
Object {
  "disableDate": [Function],
  "endDate": 2021-02-12T00:00:00.000Z,
  "highlightDate": [Function],
  "onDateSet": [Function],
  "precision": "month",
  "renderAs": [Function],
  "selectedTimestamp": 1578787200000,
  "startDate": 2019-02-12T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578787200000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: year  1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Header
    </div>
    <div>
      Selector
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: year  2`] = `
Object {
  "renderAs": [Function],
  "title": "",
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: year  3`] = `
Object {
  "endDate": 2021-02-12T00:00:00.000Z,
  "precision": "year",
  "renderAs": [Function],
  "selectedTimestamp": 1578787200000,
  "startDate": 2019-02-12T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578787200000,
}
`;

exports[`Datepicker renders date related props: initialDate: Sun Jan 12 2020 00:00:00 GMT+0000 (Coordinated Universal Time), startDate: Tue Feb 12 2019 00:00:00 GMT+0000 (Coordinated Universal Time), endDate: Fri Feb 12 2021 00:00:00 GMT+0000 (Coordinated Universal Time), today: Mon Jan 13 2020 00:00:00 GMT+0000 (Coordinated Universal Time), minPrecision: year  4`] = `
Object {
  "disableDate": [Function],
  "endDate": 2021-02-12T00:00:00.000Z,
  "highlightDate": [Function],
  "onDateSet": [Function],
  "precision": "year",
  "renderAs": [Function],
  "selectedTimestamp": 1578787200000,
  "startDate": 2019-02-12T00:00:00.000Z,
  "todayTimestamp": 1578614400000,
  "visibleTimestamp": 1578787200000,
  "wrapperClassName": "",
}
`;

exports[`Datepicker renders title 1`] = `
Object {
  "renderAs": [Function],
  "title": "Title",
}
`;

exports[`Datepicker renders without header 1`] = `
<DocumentFragment>
  <div
    aria-label=""
    class="datepickerWrapper"
    role="dialog"
    style="--fontPrimary: Helvetica Neue; --fontSecondary: Arial; --fontFallback: sans-serif; --fontSize: 14px; --fontWeight: 400; --fontSizeSmall: 10px; --fontSizeBig: 24px; --calendarWidth: 240px; --datePickerWidth: calc(var(--calendarWidth) + var(--innerPadding) * 2); --selectorHeight: 36px; --calendarHeight: 200px; --innerPadding: 12px; --entryHeight: 32px; --entryMargin: 1px; --datepickerBorderRadius: 4px; --entryBorderRadius: 4px; --bgColor: white; --textColor: black; --headerBgColor: #43a047; --weekendTextColor: red; --headerTextColor: white; --borderColor: rgba(0, 0, 0, 0.38); --modalBgColor: rgba(0, 0, 0, 0.38); --weekDayColor: rgba(0, 0, 0, 0.38); --entryHoverColor: rgba(0, 0, 0, 0.08); --entrySelectedColor: #43a047; --entryHighlightColor: lightblue; --inputWidth: 90px; --inputPadding: 8px; --inputBorderRadius: 4px; --inputBorderColor: rgba(0, 0, 0, 0.38); --inputHighlightColor: lightblue;"
  >
    <div>
      Selector
    </div>
    <div>
      Weekdays
    </div>
    <div>
      Calendar
    </div>
  </div>
</DocumentFragment>
`;
